{
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "orig_nbformat": 4,
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.8.5 64-bit ('.venv': venv)"
  },
  "interpreter": {
   "hash": "f7e9aa0a9fe1ceb3c33c5aa0b358d6952771ff9c315b2715a747dc81bd71c7e9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2,
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "import numpy as np  \n",
    "import pandas as pd  \n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.model_selection import RepeatedStratifiedKFold\n",
    "!pip install xgboost\n",
    "from xgboost import XGBClassifier\n",
    "import pyarrow.feather as feather\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.metrics import confusion_matrix\n",
    "from sklearn.model_selection import train_test_split\n",
    "# Suppress warnings \n",
    "# (sometimes you might want to ignore warnings, that's how you can achieve this)\n",
    "import warnings\n",
    "import shap\n",
    "warnings.filterwarnings('ignore')\n",
    "RSEED= 42"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Requirement already satisfied: xgboost in /Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages (1.4.2)\n",
      "Requirement already satisfied: numpy in /Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages (from xgboost) (1.21.0)\n",
      "Requirement already satisfied: scipy in /Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages (from xgboost) (1.6.0)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "export_df = feather.read_feather(\"../data/cleaned_data.feather\")"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "# Function to split the dataset \n",
    "def splitdataset(df):\n",
    "    y=export_df[\"interesting_message\"]\n",
    "    X=export_df.drop(\"interesting_message\",axis=1)\n",
    "    X_train, X_test, y_train, y_test = train_test_split(X,y, stratify = y, test_size = 0.2,random_state = 42)\n",
    "    print(y_train.dtypes)\n",
    "    return X_train, X_test, y_train, y_test"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "source": [
    "def train_xgb(X_train, X_test, y_train): \n",
    "    # Creating the classifier object \n",
    "    xgb_class = XGBClassifier(use_label_encoder=False,eval_metric= \"logloss\",enable_categorical=True)\n",
    "    xgb_parametering= {'scale_pos_weight': [1, 2, 4]}\n",
    "    cv = RepeatedStratifiedKFold(n_splits=5, n_repeats=3, random_state=1)\n",
    "    xgb_grid = GridSearchCV(estimator=xgb_class, param_grid=xgb_parametering, cv=cv,scoring='balanced_accuracy',\n",
    "    verbose=10, n_jobs=-1)\n",
    "    grid_result=xgb_grid.fit(X_train,y_train)\n",
    "    best_model=grid_result.best_estimator_\n",
    "    # report the best configuration\n",
    "    print(\"Best: %f using %s\" % (grid_result.best_score_, grid_result.best_params_))\n",
    "    return best_model\n",
    "    #TODO ADD SHAP"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "source": [
    "# Function to make predictions \n",
    "def prediction(X_test, model): \n",
    "    y_pred = model.predict(X_test)\n",
    "    return y_pred "
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "source": [
    "# Function to calculate accuracy \n",
    "def class_metrics(y_test, y_pred): \n",
    "    accuracy = balanced_accuracy_score(y_test, y_pred)\n",
    "    cm = confusion_matrix(y_test, y_pred).round()\n",
    "    print(\"Predicted values:\\n\", y_pred) \n",
    "    print(\"Confusion Matrix: \\n\", cm) \n",
    "    print(\"Balanced Accuracy: %.4f%%\" % (accuracy * 100.0))\n",
    "    print(\"Report : \\n\", classification_report(y_test, y_pred))\n",
    "    return cm, accuracy\n",
    "    #cm = confusion_matrix(y_test, y_pred).round()\n",
    "    #print(\"Predicted values:\\n\", y_pred) \n",
    "    #print(\"Confusion Matrix: \\n\", cm) \n",
    "    #scores = cross_val_score(xgb_class, y_test, y_pred, scoring='balanced_accuracy', cv=cv, n_jobs=-1)\n",
    "    #print(\"Report : \\n\", scores)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "# Driver code \n",
    "def main(): \n",
    "    # Building Phase \n",
    "    X_train, X_test, y_train, y_test = splitdataset(export_df) \n",
    "    basemodel = train_xgb(X_train, X_test, y_train)\n",
    "    # Operational Phase \n",
    "    print(\"-----\"*15)\n",
    "    print(\"Results:\\n\")\n",
    "    # Prediction\n",
    "    y_pred = prediction(X_test, basemodel) \n",
    "    cm,accuracy=class_metrics(y_test, y_pred)\n",
    "    return basemodel,y_pred,cm,accuracy"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "source": [
    "basemodel,y_pred,cm,accuracy=main()"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "int64\n",
      "Fitting 15 folds for each of 3 candidates, totalling 45 fits\n",
      "[CV 1/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 2/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 5/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 4/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 3/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 8/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 6/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 7/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 1/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   0.9s\n",
      "[CV 4/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.0s\n",
      "[CV 6/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.0s\n",
      "[CV 8/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.0s\n",
      "[CV 2/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.1s\n",
      "[CV 3/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.1s\n",
      "[CV 7/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.1s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 5/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   1.1s\n",
      "[CV 9/15; 1/3] START scale_pos_weight=1.........................................\n",
      "[CV 10/15; 1/3] START scale_pos_weight=1........................................\n",
      "[CV 11/15; 1/3] START scale_pos_weight=1........................................\n",
      "[CV 12/15; 1/3] START scale_pos_weight=1........................................\n",
      "[CV 9/15; 1/3] END ............scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 13/15; 1/3] START scale_pos_weight=1........................................\n",
      "[CV 10/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 11/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 14/15; 1/3] START scale_pos_weight=1........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 15/15; 1/3] START scale_pos_weight=1........................................\n",
      "[CV 12/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 13/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 1/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 2/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 14/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 3/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 4/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 15/15; 1/3] END ...........scale_pos_weight=1;, score=nan total time=   0.3s\n",
      "[CV 5/15; 2/3] START scale_pos_weight=2.........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 1/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.4s\n",
      "[CV 6/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 7/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 8/15; 2/3] START scale_pos_weight=2.........................................\n",
      "[CV 2/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.4s\n",
      "[CV 9/15; 2/3] START scale_pos_weight=2.........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 3/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.5s\n",
      "[CV 10/15; 2/3] START scale_pos_weight=2........................................\n",
      "[CV 11/15; 2/3] START scale_pos_weight=2........................................\n",
      "[CV 4/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.6s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 5/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.6s\n",
      "[CV 6/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.6s\n",
      "[CV 8/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.5s\n",
      "[CV 7/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.5s\n",
      "[CV 9/15; 2/3] END ............scale_pos_weight=2;, score=nan total time=   0.6s\n",
      "[CV 12/15; 2/3] START scale_pos_weight=2........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 10/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.5s\n",
      "[CV 11/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.4s\n",
      "[CV 13/15; 2/3] START scale_pos_weight=2........................................\n",
      "[CV 14/15; 2/3] START scale_pos_weight=2........................................\n",
      "[CV 15/15; 2/3] START scale_pos_weight=2........................................\n",
      "[CV 1/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 2/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 12/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.3s\n",
      "[CV 3/15; 3/3] START scale_pos_weight=4.........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 4/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 5/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 13/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.4s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 6/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 14/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.4s\n",
      "[CV 15/15; 2/3] END ...........scale_pos_weight=2;, score=nan total time=   0.5s\n",
      "[CV 2/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 1/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 3/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 4/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 5/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 7/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 8/15; 3/3] START scale_pos_weight=4.........................................\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 6/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 9/15; 3/3] START scale_pos_weight=4.........................................\n",
      "[CV 10/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 11/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 12/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 14/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 13/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 7/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 8/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.5s\n",
      "[CV 15/15; 3/3] START scale_pos_weight=4........................................\n",
      "[CV 9/15; 3/3] END ............scale_pos_weight=4;, score=nan total time=   0.6s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 10/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.6s\n",
      "[CV 12/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.6s\n",
      "[CV 13/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.6s\n",
      "[CV 14/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.6s\n",
      "[CV 11/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.7s\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1158, in fit\n",
      "    train_dmatrix, evals = _wrap_evaluation_matrices(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 236, in _wrap_evaluation_matrices\n",
      "    train_dmatrix = create_dmatrix(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\", line 1172, in <lambda>\n",
      "    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 436, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\", line 541, in __init__\n",
      "    handle, feature_names, feature_types = dispatch_data_backend(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 573, in dispatch_data_backend\n",
      "    return _from_pandas_df(data, enable_categorical, missing, threads,\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 258, in _from_pandas_df\n",
      "    data, feature_names, feature_types = _transform_pandas_df(\n",
      "  File \"/Users/fortes/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\", line 223, in _transform_pandas_df\n",
      "    raise ValueError(msg + ', '.join(bad_fields))\n",
      "ValueError: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n",
      "                categorical type is supplied, DMatrix parameter\n",
      "                `enable_categorical` must be set to `True`.lenght\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[CV 15/15; 3/3] END ...........scale_pos_weight=4;, score=nan total time=   0.5s\n"
     ]
    },
    {
     "output_type": "error",
     "ename": "ValueError",
     "evalue": "DataFrame.dtypes for data must be int, float, bool or categorical.  When\n                categorical type is supplied, DMatrix parameter\n                `enable_categorical` must be set to `True`.lenght",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/84/vkl81xdd38l550lfwl8_nngm0000gn/T/ipykernel_16618/3532969934.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mbasemodel\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my_pred\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcm\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maccuracy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/var/folders/84/vkl81xdd38l550lfwl8_nngm0000gn/T/ipykernel_16618/3679280773.py\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;31m# Building Phase\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mX_train\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mX_test\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_train\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_test\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msplitdataset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexport_df\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0mbasemodel\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtrain_xgb\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX_train\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mX_test\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_train\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0;31m# Operational Phase\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"-----\"\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0;36m15\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/var/folders/84/vkl81xdd38l550lfwl8_nngm0000gn/T/ipykernel_16618/2578718784.py\u001b[0m in \u001b[0;36mtrain_xgb\u001b[0;34m(X_train, X_test, y_train)\u001b[0m\n\u001b[1;32m      6\u001b[0m     xgb_grid = GridSearchCV(estimator=xgb_class, param_grid=xgb_parametering, cv=cv,scoring='balanced_accuracy',\n\u001b[1;32m      7\u001b[0m     verbose=10, n_jobs=-1)\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mgrid_result\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mxgb_grid\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX_train\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my_train\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     \u001b[0mbest_model\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgrid_result\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0;31m# report the best configuration\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36minner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     61\u001b[0m             \u001b[0mextra_args\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mall_args\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mextra_args\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 63\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m             \u001b[0;31m# extra_args > 0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/sklearn/model_selection/_search.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y, groups, **fit_params)\u001b[0m\n\u001b[1;32m    878\u001b[0m             \u001b[0mrefit_start_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    879\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0my\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 880\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    881\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    882\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\u001b[0m in \u001b[0;36minner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 436\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0minner_f\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y, sample_weight, base_margin, eval_set, eval_metric, early_stopping_rounds, verbose, xgb_model, sample_weight_eval_set, base_margin_eval_set, feature_weights, callbacks)\u001b[0m\n\u001b[1;32m   1156\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Please reshape the input data X into 2-dimensional matrix.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1157\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1158\u001b[0;31m         train_dmatrix, evals = _wrap_evaluation_matrices(\n\u001b[0m\u001b[1;32m   1159\u001b[0m             \u001b[0mmissing\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmissing\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1160\u001b[0m             \u001b[0mX\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\u001b[0m in \u001b[0;36m_wrap_evaluation_matrices\u001b[0;34m(missing, X, y, group, qid, sample_weight, base_margin, feature_weights, eval_set, sample_weight_eval_set, base_margin_eval_set, eval_group, eval_qid, create_dmatrix, label_transform)\u001b[0m\n\u001b[1;32m    234\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    235\u001b[0m     \"\"\"\n\u001b[0;32m--> 236\u001b[0;31m     train_dmatrix = create_dmatrix(\n\u001b[0m\u001b[1;32m    237\u001b[0m         \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    238\u001b[0m         \u001b[0mlabel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlabel_transform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/sklearn.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(**kwargs)\u001b[0m\n\u001b[1;32m   1170\u001b[0m             \u001b[0meval_group\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1171\u001b[0m             \u001b[0meval_qid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1172\u001b[0;31m             \u001b[0mcreate_dmatrix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mDMatrix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnthread\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mn_jobs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1173\u001b[0m             \u001b[0mlabel_transform\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlabel_transform\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1174\u001b[0m         )\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\u001b[0m in \u001b[0;36minner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marg\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marg\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 436\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0minner_f\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/core.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, label, weight, base_margin, missing, silent, feature_names, feature_types, nthread, group, qid, label_lower_bound, label_upper_bound, feature_weights, enable_categorical)\u001b[0m\n\u001b[1;32m    539\u001b[0m         \u001b[0;32mfrom\u001b[0m \u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mdispatch_data_backend\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    540\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 541\u001b[0;31m         handle, feature_names, feature_types = dispatch_data_backend(\n\u001b[0m\u001b[1;32m    542\u001b[0m             \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    543\u001b[0m             \u001b[0mmissing\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmissing\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\u001b[0m in \u001b[0;36mdispatch_data_backend\u001b[0;34m(data, missing, threads, feature_names, feature_types, enable_categorical)\u001b[0m\n\u001b[1;32m    571\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0m_from_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmissing\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_names\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_types\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    572\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0m_is_pandas_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 573\u001b[0;31m         return _from_pandas_df(data, enable_categorical, missing, threads,\n\u001b[0m\u001b[1;32m    574\u001b[0m                                feature_names, feature_types)\n\u001b[1;32m    575\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0m_is_pandas_series\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\u001b[0m in \u001b[0;36m_from_pandas_df\u001b[0;34m(data, enable_categorical, missing, nthread, feature_names, feature_types)\u001b[0m\n\u001b[1;32m    256\u001b[0m def _from_pandas_df(data, enable_categorical, missing, nthread,\n\u001b[1;32m    257\u001b[0m                     feature_names, feature_types):\n\u001b[0;32m--> 258\u001b[0;31m     data, feature_names, feature_types = _transform_pandas_df(\n\u001b[0m\u001b[1;32m    259\u001b[0m         data, enable_categorical, feature_names, feature_types)\n\u001b[1;32m    260\u001b[0m     return _from_numpy_array(data, missing, nthread, feature_names,\n",
      "\u001b[0;32m~/neuefische/Capstone-WIMT/.venv/lib/python3.8/site-packages/xgboost/data.py\u001b[0m in \u001b[0;36m_transform_pandas_df\u001b[0;34m(data, enable_categorical, feature_names, feature_types, meta, meta_type)\u001b[0m\n\u001b[1;32m    221\u001b[0m                 \u001b[0mcategorical\u001b[0m \u001b[0mtype\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0msupplied\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mDMatrix\u001b[0m \u001b[0mparameter\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m                 `enable_categorical` must be set to `True`.\"\"\"\n\u001b[0;32m--> 223\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m', '\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbad_fields\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    225\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mfeature_names\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mmeta\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: DataFrame.dtypes for data must be int, float, bool or categorical.  When\n                categorical type is supplied, DMatrix parameter\n                `enable_categorical` must be set to `True`.lenght"
     ]
    }
   ],
   "metadata": {
    "tags": []
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "f = open('ml-log_xgboost.txt', 'a')\n",
    "f.write('Base model: XGBoost\\n Predicted values:\\n {}\\n Confusion Matrix:\\n {}\\n Balanced Accuracy:\\n {} \\n Model: {}'.format(y_pred,cm,accuracy,basemodel))\n",
    "f.close()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [],
   "metadata": {}
  }
 ]
}